src/Diagrams/TwoD/Path/Boolean.hs

```diff
@@ -241,6 +244,32 @@ loopExclusion :: Double -> FillRule
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
-loopExclusion tol fill path1 path2 =
+loopExclusion tol fill path1 path2 = normalizeWindings fill $
   map path2loop $ C.exclusion (map loop2path path1)
   (map loop2path path2) (fillrule fill) tol
+
+-- Force all top level loops to wind counterclockwise and revese inner loops as needed.
+normalizeWindings :: FillRule -> [Located (Trail' Loop V2 Double)] -> [Located (Trail' Loop V2 Double)]
+normalizeWindings fill = concat . map forceCC . nestedGroups fill where
+  forceCC ls | (l:_) <- ls, isClockwise l = map reverseLocLoop ls
+             | otherwise                  = ls
+
+-- Group a list of loops such that the first element of each group contains all the others.
+nestedGroups :: FillRule -> [Located (Trail' Loop V2 Double)] -> [[Located (Trail' Loop V2 Double)]]
+nestedGroups fill = map flatten . go [] where
+  go ts               []                      = ts
+  go []               (l:ls)                  = go [Node l []]                      ls
+  go ((Node n ns):ts) (l:ls) | l `contains` n = go [Node l [Node n ns]]             ls
+                             | n `contains` l = go (Node n (go ns [l]) : ts)        ls
+                             | otherwise      = go (Node n ns          : go ts [l]) ls
+  contains = containsBy fill
+
+-- To test if s contains t, we can merely test if any point of t is within s
+-- because the binary operations guarantee that the loops do not intersect.
+containsBy :: FillRule -> Located (Trail' Loop V2 Double) -> Located (Trail' Loop V2 Double) -> Bool
+containsBy Winding s t = isInsideWinding s (atStart t)
+containsBy EvenOdd s t = isInsideEvenOdd s (atStart t)
+
+-- Test if a loop winds clockwise.
```

Comment:
Thanks for the clarification! Lets add this to the comment and somewhere we should add a test:

```haskell
prop_sample_start :: Located (Trail' Loop V2 Double) -> Bool
prop_sample_start l = sample l (atStart l) /= 0
```

```suggestion
-- Test if a loop winds clockwise.
-- Points that are precisely on a loop as well as those inside the loop will
-- have non-zero winding number. The winding number of points in a loop
-- will be positive if the loop is counterclockwise and negative if the loop
-- is clockwise.
```

Reply:
I'll add the comment, but the prop isn't true because in the case that `l` contains zero area `sample l (atStart l) == 0`.

Reply:
Thinking about it more, I'm not really sure how reliably `sample l (atStart l) /= 0`. I don't entirely trust the code that finds winding numbers ([see diagrams-lib #337](https://github.com/diagrams/diagrams-lib/issues/337)). I can add a redundant check in the case that `sample l (atStart l) == 0`.